### DISCUSSION TITLE
Added HOUDINI_OTLSCAN_PATH to system environment variable, but got error when I started up Houdini 19.5

### DISCUSSION THREAD

**Discussion by supwang** [stassius/StableHoudini#16] [Q&A] [answered]
I added HOUDINI_OTLSCAN_PATH to system environment variable and put the path to the assets folder there, but got below error message when I started up Houdini 19.5.
Anyone knows how to solve? Thank you.

opalias: 'mountain' is not a known operator.
opalias: 'ripple' is not a known operator.
opalias: 'pointmap' is not a known operator.
opalias: 'waveform' is not a known operator.
opalias: 'grainsource' is not a known operator.
opalias: 'baketexture' is not a known operator.
opalias: 'guidegroom' is not a known operator.
opalias: 'guidetransfer' is not a known operator.
opalias: 'guidedraw' is not a known operator.
opalias: 'guideprocess' is not a known operator.
opalias: 'guideinit' is not a known operator.
opalias: 'hairclump' is not a known operator.
opalias: 'hairgen' is not a known operator.
opalias: 'volumevelocityfromcurves' is not a known operator.
opalias: 'volumevelocityfromsurface' is not a known operator.
opalias: 'femsolver' is not a known operator.
opalias: 'femsolver' is not a known operator.
opalias: 'femsolidobject' is not a known operator.
opalias: 'femsolidconfigureobject' is not a known operator.
opalias: 'femhybridobject' is not a known operator.
opalias: 'femhybridconfigureobject' is not a known operator.
opalias: 'femfuseconstraint' is not a known operator.
opalias: 'femregionconstraint' is not a known operator.
opalias: 'femtargetconstraint' is not a known operator.
opalias: 'femattachconstraint' is not a known operator.
opalias: 'femvalidate' is not a known operator.
opalias: 'femvalidate' is not a known operator.
opalias: 'tetembed' is not a known operator.
opalias: 'tetconform' is not a known operator.
opalias: 'tetfracture' is not a known operator.
opalias: 'pyrosolver::2.0' is not a known operator.
opalias: 'smokesolver::2.0' is not a known operator.
opalias: 'gasupres::2.0' is not a known operator.
opalias: 'flipsolver::2.0' is not a known operator.
opalias: 'popsolver::2.0' is not a known operator.
opalias: 'popwrangle' is not a known operator.
opalias: 'popcollisionbehavior' is not a known operator.
opalias: 'impactanalysis' is not a known operator.
opalias: 'attribblur' is not a known operator.
opalias: 'heightfield_quickshade' is not a known operator.
opalias: 'planarpatch' is not a known operator.
opalias: 'planarpatchfromcurves' is not a known operator.
opalias: 'autoselectlod' is not a known operator.
opalias: 'retimeinstances' is not a known operator.
opalias: 'extractinstances' is not a known operator.
opalias: 'simulationproxy' is not a known operator.
opalias: 'assignprototypes' is not a known operator.
opalias: 'attribfrompieces' is not a known operator.
opalias: 'attribfrompieces' is not a known operator.
opalias: 'scatteralign' is not a known operator.
opalias: 'maskbyfeature' is not a known operator.
opalias: 'flipmodifyvolume' is not a known operator.
opalias: 'flipboundary' is not a known operator.
opalias: 'pxrfractalize' is not a known operator.
opalias: 'fit' is not a known operator.
opalias: 'fibergroom' is not a known operator.
opalias: 'volumeshadercore' is not a known operator.
opalias: 'brushedmetalshadercore' is not a known operator.
opalias: 'carpaintshadercore' is not a known operator.
opalias: 'skinshadercore' is not a known operator.
opalias: 'pythonserver' is not a known operator.
opalias: 'kinefx::adapttoterrain' is not a known operator.
opalias: 'kinefx::fbikconfiguretargets' is not a known operator.
opalias: 'kinefx::rigdoctor' is not a known operator.
opalias: 'kinefx::rigmirrorpose' is not a known operator.
opalias: 'kinefx::rigpose' is not a known operator.
opalias: 'kinefx::skeleton' is not a known operator.
opalias: 'kinefx::parentjoints' is not a known operator.
opalias: 'kinefx::orientjoints' is not a known operator.
opalias: 'kinefx::configurejoints' is not a known operator.
opalias: 'kinefx::jointcapturebiharmonic' is not a known operator.
opalias: 'labs::gaea_tor_processor' is not a known operator.
opalias: 'labs::simple_retime' is not a known operator.
opalias: 'labs::coord_swizzle_vector::1.0' is not a known operator.
opalias: 'labs::dds_file' is not a known operator.
opalias: 'labs::normal_color' is not a known operator.
opalias: 'labs::normal_combine' is not a known operator.
opalias: 'labs::normal_invert' is not a known operator.
opalias: 'labs::normal_levels' is not a known operator.
opalias: 'labs::normal_map' is not a known operator.
opalias: 'labs::normal_normalize' is not a known operator.
opalias: 'labs::normal_rotate' is not a known operator.
opalias: 'labs::curve_branches' is not a known operator.
opalias: 'labs::color_blend' is not a known operator.
opalias: 'labs::dissolve_flat_edges' is not a known operator.
opalias: 'labs::osm_filter' is not a known operator.
opalias: 'labs::sine_wave' is not a known operator.
opalias: 'labs::spiral' is not a known operator.
opalias: 'labs::straight_skeleton_2D' is not a known operator.
opalias: 'labs::labs_shader' is not a known operator.
opalias: 'labs::toon_glsl_shader' is not a known operator.
opalias: 'labs::multi_bounding_box' is not a known operator.
opalias: 'labs::fire_presets' is not a known operator.
opalias: 'labs::impostor_camera_rig' is not a known operator.
opalias: 'labs::lod_hierarchy' is not a known operator.
opalias: 'labs::3d_facebook_image' is not a known operator.
opalias: 'labs::csv_exporter' is not a known operator.
opalias: 'labs::games_baker::2.0' is not a known operator.
opalias: 'labs::games_baker' is not a known operator.
opalias: 'labs::impostor_texture' is not a known operator.
opalias: 'labs::marmoset_export' is not a known operator.
opalias: 'labs::rbd_to_fbx' is not a known operator.
opalias: 'labs::sketchfab' is not a known operator.
opalias: 'labs::vector_field' is not a known operator.
opalias: 'labs::vertex_animation_textures::2.0' is not a known operator.
opalias: 'labs::vertex_animation_textures' is not a known operator.
opalias: 'labs::volume_texture' is not a known operator.
opalias: 'labs::flowmap_shader' is not a known operator.
opalias: 'labs::impostor_turntable_lens' is not a known operator.
opalias: 'labs::motionvector_shader' is not a known operator.
opalias: 'labs::octahedron_impostor_lens' is not a known operator.
opalias: 'labs::align_and_distribute' is not a known operator.
opalias: 'labs::autouv' is not a known operator.
opalias: 'labs::av_analyze_images' is not a known operator.
opalias: 'labs::av_analyze_images::2.0' is not a known operator.
opalias: 'labs::av_analyze_images::3.0' is not a known operator.
opalias: 'labs::av_depth_map' is not a known operator.
opalias: 'labs::av_depth_map::2.0' is not a known operator.
opalias: 'labs::av_depth_map::3.0' is not a known operator.
opalias: 'labs::av_initialize' is not a known operator.
opalias: 'labs::av_initialize::2.0' is not a known operator.
opalias: 'labs::av_initialize::3.0' is not a known operator.
opalias: 'labs::av_meshing' is not a known operator.
opalias: 'labs::av_meshing::2.0' is not a known operator.
opalias: 'labs::av_meshing::3.0' is not a known operator.
opalias: 'labs::av_photogrammetry' is not a known operator.
opalias: 'labs::av_photogrammetry::2.0' is not a known operator.
opalias: 'labs::av_photogrammetry::3.0' is not a known operator.
opalias: 'labs::av_structure_from_motion' is not a known operator.
opalias: 'labs::av_structure_from_motion::2.0' is not a known operator.
opalias: 'labs::av_structure_from_motion::3.0' is not a known operator.
opalias: 'labs::av_texturing' is not a known operator.
opalias: 'labs::av_texturing::2.0' is not a known operator.
opalias: 'labs::av_texturing::3.0' is not a known operator.
opalias: 'labs::box_clip' is not a known operator.
opalias: 'labs::building_generator::4.0' is not a known operator.
opalias: 'labs::building_generator_utility' is not a known operator.
opalias: 'labs::cable_generator' is not a known operator.
opalias: 'labs::calculate_slope' is not a known operator.
opalias: 'labs::cylinder_generator' is not a known operator.
opalias: 'labs::decal_projector' is not a known operator.
opalias: 'labs::delete_small_parts' is not a known operator.
opalias: 'labs::delight' is not a known operator.
opalias: 'labs::destruction_cleanup' is not a known operator.
opalias: 'labs::destruction_cleanup::2.0' is not a known operator.
opalias: 'labs::detail_mesh' is not a known operator.
opalias: 'labs::dirtskirt' is not a known operator.
opalias: 'labs::disc_generator' is not a known operator.
opalias: 'labs::edgegroup_to_curve' is not a known operator.
opalias: 'labs::edge_color' is not a known operator.
opalias: 'labs::extract_borders' is not a known operator.
opalias: 'labs::extract_filename' is not a known operator.
opalias: 'labs::extract_silhouette::1.0' is not a known operator.
opalias: 'labs::flowmap' is not a known operator.
opalias: 'labs::flowmap::2.0' is not a known operator.
opalias: 'labs::flowmap_guide' is not a known operator.
opalias: 'labs::flowmap_obstacle' is not a known operator.
opalias: 'labs::flowmap_obstacle::2.0' is not a known operator.
opalias: 'labs::flowmap_obstacle::3.0' is not a known operator.
opalias: 'labs::flowmap_to_color' is not a known operator.
opalias: 'labs::flowmap_visualize' is not a known operator.
opalias: 'labs::gaea_tor_processor' is not a known operator.
opalias: 'labs::gaea_tor_processor::2.0' is not a known operator.
opalias: 'labs::gameres::2.0' is not a known operator.
opalias: 'labs::gameres' is not a known operator.
opalias: 'labs::goz_export' is not a known operator.
opalias: 'labs::goz_import' is not a known operator.
opalias: 'labs::group_by_attribute' is not a known operator.
opalias: 'labs::group_curve_corners' is not a known operator.
opalias: 'labs::inside_face_uvs' is not a known operator.
opalias: 'labs::instant_meshes' is not a known operator.
opalias: 'labs::instant_meshes::2.0' is not a known operator.
opalias: 'labs::lod_create::2.0' is not a known operator.
opalias: 'labs::lod_create' is not a known operator.
opalias: 'labs::lot_subdivision' is not a known operator.
opalias: 'labs::make_loop' is not a known operator.
opalias: 'labs::mapbox' is not a known operator.
opalias: 'labs::maps_baker' is not a known operator.
opalias: 'labs::maps_baker::2.0' is not a known operator.
opalias: 'labs::maps_baker::3.0' is not a known operator.
opalias: 'labs::measure_curvarture' is not a known operator.
opalias: 'labs::measure_curvarture::2.0' is not a known operator.
opalias: 'labs::merge_small_islands' is not a known operator.
opalias: 'labs::mesh_slice' is not a known operator.
opalias: 'labs::mesh_tiler' is not a known operator.
opalias: 'labs::min_max_average' is not a known operator.
opalias: 'labs::niagara' is not a known operator.
opalias: 'labs::obj_importer' is not a known operator.
opalias: 'labs::osm_buildings' is not a known operator.
opalias: 'labs::osm_import' is not a known operator.
opalias: 'labs::path_deform' is not a known operator.
opalias: 'labs::physics_painter' is not a known operator.
opalias: 'labs::polydeform' is not a known operator.
opalias: 'labs::procedural_smoke' is not a known operator.
opalias: 'labs::quadrangulate' is not a known operator.
opalias: 'labs::quickmaterial' is not a known operator.
opalias: 'labs::quickmaterial::2.0' is not a known operator.
opalias: 'labs::rbd_edge_strip' is not a known operator.
opalias: 'labs::remove_inside_faces' is not a known operator.
opalias: 'labs::repair' is not a known operator.
opalias: 'labs::simple_retime' is not a known operator.
opalias: 'labs::scifi_panels' is not a known operator.
opalias: 'labs::simple_baker::2.0' is not a known operator.
opalias: 'labs::simple_baker' is not a known operator.
opalias: 'labs::sketchfab_output' is not a known operator.
opalias: 'labs::skinning_converter' is not a known operator.
opalias: 'labs::snow_buildup' is not a known operator.
opalias: 'labs::soften_normals' is not a known operator.
opalias: 'labs::sphere_generator' is not a known operator.
opalias: 'labs::static_fracture_export' is not a known operator.
opalias: 'labs::straighten' is not a known operator.
opalias: 'labs::straight_skeleton_3d' is not a known operator.
opalias: 'labs::sweep_geometry' is not a known operator.
opalias: 'labs::terrain_layer_export' is not a known operator.
opalias: 'labs::terrain_layer_import' is not a known operator.
opalias: 'labs::terrain_segment' is not a known operator.
opalias: 'labs::terrain_texture' is not a known operator.
opalias: 'labs::thicken' is not a known operator.
opalias: 'labs::trace_psd_file' is not a known operator.
opalias: 'labs::trace_psd_file::2.0' is not a known operator.
opalias: 'labs::turntable' is not a known operator.
opalias: 'labs::unreal_pivotpainter' is not a known operator.
opalias: 'labs::uv_transfer' is not a known operator.
opalias: 'labs::uv_unitize' is not a known operator.
opalias: 'labs::visualize_uvs' is not a known operator.
opalias: 'labs::coord_swizzle_quaternion::1.0' is not a known operator.
opalias: 'labs::exoside_quadremesher' is not a known operator.
opalias: 'labs::distance_from_border' is not a known operator.
opalias: 'labs::quadrangulate::2.0' is not a known operator.
opalias: 'labs::symmetrize' is not a known operator.
opalias: 'labs::rizomuv_rectangularize' is not a known operator.
opalias: 'labs::rizomuv_optimize' is not a known operator.
opalias: 'labs::rizomuv_unwrap' is not a known operator.
opalias: 'labs::rizomuv_processor' is not a known operator.
Warning: Bad operator type when binding handles: Object/path
Warning: Bad operator type when binding handles: Object/pathcv
Warning: Bad operator type when binding handles: Object/cop2plane
Warning: Bad operator type when binding handles: Sop/agentcollisionlayer
Warning: Bad operator type when binding handles: Sop/agentconfigurejoints
Warning: Bad operator type when binding handles: Sop/agentedit
Warning: Bad operator type when binding handles: Sop/attribadjustfloat
Warning: Bad operator type when binding handles: Sop/attribadjustcolor
Warning: Bad operator type when binding handles: Sop/attribadjustvector
Warning: Bad operator type when binding handles: Sop/attribadjustinteger
Warning: Bad operator type when binding handles: Sop/attribwrangle
Warning: Bad operator type when binding handles: Sop/attribexpression
Warning: Bad operator type when binding handles: Sop/convertline
Warning: Bad operator type when binding handles: Sop/crosssectionsurface
Warning: Bad operator type when binding handles: Sop/explodedview::2.0
Warning: Bad operator type when binding handles: Sop/falloff
Warning: Bad operator type when binding handles: Sop/matchsize
Warning: Bad operator type when binding handles: Sop/mountain::2.0
Warning: Bad operator type when binding handles: Sop/ripple::2.0
Warning: Bad operator type when binding handles: Sop/pointwrangle
Warning: Bad operator type when binding handles: Sop/revolve::2.0
Warning: Bad operator type when binding handles: Sop/split
Warning: Bad operator type when binding handles: Cop2/emboss

**Comment by stassius** [comment, marked as answer]
You also need to add @/otls as an entry to this variable, so it scans the home directory as well.
Like this: 

![image](https://user-images.githubusercontent.com/35260274/235700340-9faeb95e-18d1-4079-a65b-9fa158cfcfe0.png)
So the whole variable would look like this: path/to/your/hda;@/otls

**Reply by supwang** [reply]
Thank you.
But when I adding @/otls into environment, I got character error. ("/" can not be added)

<img width="299" alt="1" src="https://user-images.githubusercontent.com/4122964/235819561-4a5befdb-aa70-455c-9994-b42ac2e8f775.png">

<img width="431" alt="2" src="https://user-images.githubusercontent.com/4122964/235819567-80ce6448-2f50-42d4-9ded-3350328a7226.png">

**Comment by stassius** [comment]
Change the direction of the slash than. your environment variable will look like this: PATH\TO\ASSETS;@\otls

**Reply by supwang** [reply]
Thank you.
I added PATH\TO\ASSETS and @\otls, but still got the same error message.

**Reply by supwang** [reply]
> Change the direction of the slash than. your environment variable will look like this: PATH\TO\ASSETS;@\otls

If I open the saved project, the Stable assets can be called normally. Is this mean that we do not need to change the setting of environment variable?

**Comment by stassius** [comment]
When you install an asset library it will be saved in a project. No need to reinstall it each time you open the project. 
In order to add the library system-wide, you need to put the asset path to the environment variable as discussed earlier.

**Reply by supwang** [reply]
Thank you.
I always get error message when I open Houdini after I added path HOUDINI_OTLSCAN_PATH to system environment variable.
PATH\TO\ASSETS;@\otls should be added to system environment variable -> Path, is it correct?
<img width="436" alt="1" src="https://user-images.githubusercontent.com/4122964/236446607-2a01f61c-994d-432f-91b4-ca75e67fef0c.png">

**Comment by stassius** [comment]
You should add it to HOUDINI_OTLSCAN_PATH

**Reply by supwang** [reply]
Thank you.
If I add a new system variables HOUDINI_OTLSCAN_PATH, there are only variable name and value can be filled. How to add PATH\TO\ASSETS and @\otls in one variables. (Only default PATH can add multi variables.)
<img width="986" alt="1" src="https://user-images.githubusercontent.com/4122964/236632485-093994ce-eb28-47aa-9b28-197044095c5c.png">

**Comment by stassius** [comment]
Just divide it by a semicolon.

**Reply by supwang** [reply]
Solved. Thank you.
<img width="1231" alt="SD Houdini Asset" src="https://user-images.githubusercontent.com/4122964/236650553-9ba7bf2b-d0f3-423a-8425-0be062e3ee9c.png">
